specification/hardwaresecuritymodules/HardwareSecurityModules.Management/models.tsp

```diff
@@ -0,0 +1,851 @@
+
+  /**
+   * The reason for approval/rejection of the connection.
+   */
+  description?: string;
+
+  /**
+   * A message indicating if changes on the service provider require any updates on the consumer.
+   */
+  actionsRequired?: string;
+}
+
+/**
+ * Common fields that are returned in the response for all Azure Resource Manager resources
+ */
+model Resource {
+  /**
+   * Fully qualified resource ID for the resource. E.g. "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/{resourceProviderNamespace}/{resourceType}/{resourceName}"
+   */
+  @visibility(Lifecycle.Read)
+  id?: Azure.Core.armResourceIdentifier;
+
+  /**
+   * The name of the resource
+   */
+  @visibility(Lifecycle.Read)
+  name?: string;
+
+  /**
+   * The type of the resource. E.g. "Microsoft.Compute/virtualMachines" or "Microsoft.Storage/storageAccounts"
+   */
+  @visibility(Lifecycle.Read)
+  type?: string;
+
+  /**
+   * Azure Resource Manager metadata containing createdBy and modifiedBy information.
+   */
+  @visibility(Lifecycle.Read)
+  systemData?: SystemData;
+}
+
+/**
+ * Cloud Hsm Cluster SKU information
+ */
+model CloudHsmClusterSku {
+  /**
+   * Sku family of the Cloud HSM Cluster
+   */
+  family: CloudHsmClusterSkuFamily;
+
+  /**
+   * Sku name of the Cloud HSM Cluster
+   */
+  name: CloudHsmClusterSkuName;
+
+  /**
+   * Sku capacity
+   */
+  capacity?: int32;
+}
+
+/**
+ * Patchable properties of the Cloud HSM Cluster
+ */
+#suppress "@azure-tools/typespec-azure-resource-manager/patch-envelope" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+model CloudHsmClusterPatchParameters {
+  /**
+   * The Cloud HSM Cluster's tags
+   */
+  #suppress "@azure-tools/typespec-azure-resource-manager/arm-no-record" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+  tags?: Record<string>;
+
+  /**
+   * Managed service identity (system assigned and/or user assigned identities)
+   */
+  identity?: CommonTypes.ManagedServiceIdentity;
+}
+
+/**
+ * A private link resource.
+ */
+#suppress "@azure-tools/typespec-azure-core/composition-over-inheritance" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+model PrivateLinkResource extends Resource {
+  /**
+   * Resource properties.
+   */
+  #suppress "@azure-tools/typespec-azure-core/no-private-usage" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+  properties?: PrivateLinkResourceProperties;
+}
+
+/**
+ * Properties of a private link resource.
+ */
+model PrivateLinkResourceProperties {
+  /**
+   * The private link resource group id.
+   */
+  @visibility(Lifecycle.Read)
+  groupId?: string;
+
+  /**
+   * The private link resource required member names.
+   */
+  @visibility(Lifecycle.Read)
+  requiredMembers?: string[];
+
+  /**
+   * The private link resource private link DNS zone name.
+   */
+  requiredZoneNames?: string[];
+}
```

Comment:
Private endpoint connections being a standard contract across many RPs, should be modeled with a common types contract - can we do it?